src/modules/ethereum/ethHelper.js

```diff
@@ -60,3 +58,42 @@ export async function getNetworkDetails(web3) {
         type: networkType
     };
 }
+
+export async function processTx(tx, txName, gasEstimate) {
```

Comment:
should we use the same in the lock?

Reply:
yep. This is just a first pass to make it generic/reusable, we can revisit later. 
Anways it's handled the same way everywhere and adding async tx submission will be easier (I'm just working on it).